advanced.matrix/src/main/java/learning/solutions/advanced/matrix/engineeringLevel/ReinforcementLearner.java

```diff
@@ -115,25 +100,13 @@ public void train(Point source, Point dest) {
     @Deprecated
     public void train(Point source, Point dest, java.util.List<Point> captureExploration) {
         logger.info(" rlEngine training triggered at SCET = " + System.currentTimeMillis() + " source = " + source
-                .toString() + " destination = " + dest.toString());
-        this.destination = ReinforcementLearnerUtil.findPoint(dest, navGrid);
+                .toString());
         this.source = ReinforcementLearnerUtil.findPoint(source, navGrid);
 
-        destination.setReward(maxReward);
-        destination.setqValue(maxReward);
-
-        ReinforcementLearnerUtil.setDestinationReward(navGrid, destination, maxReward);
-        int altDestX = Integer.parseInt(mazeDefinition.getProperty("maze.environment.alternate.destination").split("," +
-                "")[0]);
-        int altDestY = Integer.parseInt(mazeDefinition.getProperty("maze.environment.alternate.destination").split("," +
-                "")[1]);
-        altDest = ReinforcementLearnerUtil.findPoint(new Point(altDestX, altDestY), navGrid);
-        ReinforcementLearnerUtil.setDestinationReward(navGrid, altDest, maxReward);
-
         for (int i = 0; i < maxIterations; i++) {
             RCell current = ReinforcementLearnerUtil.getRandomSource(navGrid);
             //episode
-            while (!current.equals(destination) && !current.equals(altDest)) {
+            while (!(current.getReward() >= maxReward)) {
```

Comment:
donot change the deprecated method. Do a search and make sure this deprecated method is not used except to test and demo.